### PR TITLE
Validate kots pull fix

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What type of PR is this?

kind/bug

#### What this PR does / why we need it:

This fixes kots pull cli to properly inject image pull secrets when using useHelmInstall true.

#### Which issue(s) this PR fixes:

Fixes NONE

#### Does this PR introduce a user-facing change?

```release-note
Fixed the `kots pull` cli command to properly inject imagePullSecrets when using Helm Charts with useHelmInstall set to true.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: pkg/pull/pull.go [hunk 1]
```diff
@@ -283,27 +275,30 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
...
-		if pullOptions.CreateAppDir {
-			renderDir = filepath.Join(pullOptions.RootDir, u.Name)
-		}
+	renderDir := pullOptions.RootDir
+	if pullOptions.CreateAppDir {
+		renderDir = filepath.Join(pullOptions.RootDir, u.Name)
+	}
 
-		newHelmCharts, err = kotsutil.LoadHelmChartsFromPath(renderDir)
+	newHelmCharts, err := kotsutil.LoadHelmChartsFromPath(renderDir)
```
Comment (emosbaugh) [inline pkg/pull/pull.go:283]: this needed to be set to properly set the UseHelmInstall option argument to the FindPrivateImages function

https://github.com/replicatedhq/kots/blob/e8c349031e2b43483431fa9e1edc5837a78c5442/pkg/pull/pull.go#L429

File: pkg/image/builder.go [hunk 2]
```diff
@@ -515,7 +515,7 @@ func IsPrivateImage(image string, dockerHubRegistry registry.RegistryOptions) (b
 			sysCtx.DockerInsecureSkipTLSVerify = types.OptionalBoolTrue
 		}
 
-		remoteImage, err := ref.NewImage(context.Background(), &sysCtx)
+		remoteImage, err := ref.NewImageSource(context.Background(), &sysCtx)
```
Comment (emosbaugh) [inline pkg/image/builder.go:518]: NewImage must check local docker or use docker user creds or something cause it wasnt finding private images on my codeserver.

File: .github/workflows/build-test.yaml [hunk 3]
```diff
@@ -42,7 +42,7 @@ jobs:
       - name: Setup Node.js environment
         uses: actions/setup-node@v2
         with:
-          node-version: '10'
+          node-version: '10.x'
```
Comment (emosbaugh) [inline .github/workflows/build-test.yaml:45]: testim throws an error about it being incompatible with node 10

File: .github/workflows/build-test.yaml [hunk 4]
```diff
@@ -654,6 +654,7 @@ jobs:
           kubectl -n default create secret generic multi-namespace-yeti-registry --type=kubernetes.io/dockerconfigjson --from-file=.dockerconfigjson=./replicated-registry-auth.json
 
           kustomize build $PWD/$APP_SLUG/overlays/midstream | kubectl apply -f -
+          kustomize build $PWD/$APP_SLUG/overlays/midstream/charts/redis | kubectl apply -f -
```
Comment (emosbaugh) [inline .github/workflows/build-test.yaml:657]: we now write additional "entrypoints" for kustomize for each useHelmInstall chart

File: pkg/image/builder.go (hunk 2 above)
Comment (divolgin) [reply pkg/image/builder.go:518]: It does! Good find.

**Review by divolgin — APPROVED** [review]

**Pull request merged**